src-js/optimizelyClient.js

```diff
@@ -0,0 +1,49 @@
+import * as optimizelySDK from '@optimizely/optimizely-sdk';
+const COOKIE_KEY = 'cci-org-analytics-id';
+
+class OptimizelyClient {
+  constructor() {
+    this.client = optimizelySDK.createInstance({
+      datafile: window.optimizelyDatafile,
+    });
+  }
+  // When the promise resolves, it can either be with null or with a variation name.
+  // This is consistent to what getVariation from the optimizely-sdk does
+  getVariationName(options) {
+    return new Promise((resolve, reject) => {
+      window.addEventListener("userDataReady", () => {
+        const userId = (window.userData && window.userData.analytics_id) ? window.userData.analytics_id : null;
+        const orgId = Cookies.get(COOKIE_KEY) ?? null;
+
+        if (!userId || !orgId) {
+          return resolve(null);
+        }
+
+        if (!options || !options.experimentKey || !options.groupExperimentName) {
```

Comment:
👍

Reply:
`nullish coalescing operator` is for `??`. What you want is called `optional-chaining` and I tried to add it but it was failing the whole build. We can try again on another PR soon though

Reply:
brainfart. no we're fine.

Reply:
ignore above comment.